# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/svg/types/scripted/SVGLength-viewport.html
# Commit: 5158484e9eab
# Full Hash: 5158484e9eabc55a48e73e68ad4b6e676d057997
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 04:12:49
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 2 - support vw, vh, vmin and vmax units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180619
# ==============================================================================

diff -r 35034282dc7c -r 5158484e9eab testing/web-platform/tests/svg/types/scripted/SVGLength-viewport.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/svg/types/scripted/SVGLength-viewport.html	Mon Sep 11 18:20:19 2023 +0000
@@ -0,0 +1,35 @@
+<!DOCTYPE HTML>
+<title>SVGLength with 'viewport' units</title>
+<link rel="help" href="https://www.w3.org/TR/SVG/types.html#InterfaceSVGLength">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="ref"></div>
+<svg>
+  <rect id="rect"/>
+</svg>
+<script>
+  const units = ['vw', 'vh', 'vmin', 'vmax'];
+
+  for (const unit of units) {
+
+    const ref = document.getElementById("ref");
+    ref.style.width=`10${unit}`;
+    const rect = document.getElementById("rect");
+    rect.setAttribute("x", `10${unit}`);
+
+    const ref_width = ref.offsetWidth;
+    let length = rect.x.baseVal;
+
+    test(() => {
+      assert_equals(length.unitType, SVGLength.SVG_LENGTHTYPE_UNKNOWN);
+      assert_approx_equals(length.value, ref_width, 0.5);
+    }, `${unit} unit in SVGLength`);
+
+    test(() => {
+      const old_value = length.value
+      length.value = ref_width * 2;
+      assert_approx_equals(length.valueInSpecifiedUnits, 20, 0.2);
+      length.value = old_value;
+    }, `Convert back to ${unit} from new user unit value`);
+  }
+</script>
